### PR TITLE
Deletenoteapiv2

### PR DESCRIPTION
**Pull request by zingales**

### CODE REVIEW

File: handlers/handlers.go [hunk 1]
```diff
@@ -300,8 +300,38 @@ func HandleNoteApiRequest(
 
 		fmt.Fprint(responseWriter, string(noteString))
 
+	case http.MethodDelete:
+
+		id, err := strconv.ParseInt(request.URL.Query().Get("id"), 10, 64)
```
Comment (ethomas2) [inline handlers/handlers.go]: We talked about this over slack, but I think "proper" rest api design is to have this in the path, not in the query param (`DELETE /notes/:id` instead of `DELETE /notes/?id=:id`)

Comment (ethomas2) [reply handlers/handlers.go]: or `DELETE /user/:user-id/notes/:note-id` if note is a nested resource of user

File: models/note.go [hunk 2]
```diff
@@ -32,3 +35,51 @@ func (db *DB) StoreNewNote(
 	}
 	return NoteId(noteId), nil
 }
+
+//  DB methods
+func (db *DB) GetUsersNotes(userId UserId) (NoteMap, error) {
```
[the remark this answers is not in the document]
Comment (ethomas2) [reply models/note.go]: After thinking about this a while longer, I decided that `rows.Next` and `rows.Scan` almost certainly don't read the whole query into memory. I dug into the psql driver code and confirmed
https://github.com/lib/pq/blob/master/conn.go#L1424

As far as I can tell each call to `rows.Next()` corresponds to a read of one data row message as defined in the Simple Query section of [psql documentation](https://www.postgresql.org/docs/9.3/static/protocol-flow.html#AEN99760).
tl;dr, if you call `rows.Next()` followed by `rows.Scan()`, you read exactly one row into memory (as expected).

So to avoid generating the whole NoteMap in memory, you could just return a stream of rows (either the rows object itself or a channel of notes). Maybe something like this.
```go
type Result {
    NoteId NoteId
    Note Note
    Err error
}

func (db *DB) GetUsersNotes(userId UserId) (<-chan Result) {

	resultChan := make(chan Result)
	go func() {
	    defer func() { close(resultChan) }
	    sqlQuery := `
		    SELECT id, author_id, content, creation_time FROM note
		    WHERE author_id = $1`
	    rows, err := db.Query(sqlQuery, int64(userId))
	    if err != nil {
		    resultChan <- NoteResult{Err:  convertPostgresError(err)}
		    return 
	    }
	    defer rows.Close()

	    for rows.Next() {
		    var tempId int64
		    tempNote := &Note{}
		    if err := rows.Scan(&tempId, &tempNote.AuthorId, &tempNote.Content, &tempNote.CreationTime); err != nil {
			  resultChan <- Result{Err: convertPostgresError(err)}
			  return
		    }
		    resultChan <- Result{NoteId: tempId, Note: tempNote}
	    }
	}()
	return resultChan
}
```

I mostly wrote that up because I find it pretty cool. But it's overly complicated for the current use case. A much better solution is to just get rid of this method altogether. It's unnecessary if you do a `delete where user_id = <user_id> and note_id = <note_id>`
